docs/maui/layouts/DockLayout.md

```diff
@@ -0,0 +1,121 @@
+---
+title: DockLayout - .NET MAUI Community Toolkit
+author: profix898
+description: The DockLayout is a layout where views can be docked to the sides (top, left, right, bottom) of the layout container.
+ms.date: 11/04/2022
+---
+
+# DockLayout
+
+`DockLayout` is a layout where views can be docked to the sides of the layout container.
+
+The image below shows how a `DockLayout` is conceptually structured. Child views are docked at one of 4 possible docking positions: *Top*, *Bottom*, *Left* or *Right* (equivalent to `DockPosition.Top`, `DockPosition.Bottom`, `DockPosition.Left`, and `DockPosition.Right`). Views that are not explicitly docked (or with `DockPosition.None`) are displayed at the center (or between *Top* / *Bottom* and *Left* / *Right* positions).
+
+![DockLayout concept](../images/layouts/DockLayout.svg)
+
+## Building a DockLayout
+
+A basic `DockLayout` can be created in XAML as shown here:
+
+### XAML
+
+```xml
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+             x:Class="MyProject.MyContentPage">
+
+    <mct:DockLayout>
+        <Button mct:DockLayout.DockPosition="Top" Text="Top" HeightRequest="50" />
+        <Button mct:DockLayout.DockPosition="Bottom" Text="Bottom" HeightRequest="70" />
+        <Button mct:DockLayout.DockPosition="Left" Text="Left" WidthRequest="80" />
+        <Button mct:DockLayout.DockPosition="Right" Text="Right" WidthRequest="90" />
+        <Button Text="Center" />
+    </mct:DockLayout>
+    
+</ContentPage>
+```
+
+For *Left* / *Right* docking, a `WidthRequest` should be specified. For *Top* / *Bottom* docking, a `HeightRequest` defines the size of the child view along the docking direction. The orthogonal directions are always calculated implicitly by the `DockLayout` manager.
+
+### C#
+
+To set the docking position from C#, use `DockLayout.SetDockPosition(IView, DockPosition)` to apply the attached `DockPosition` property.
+
+```csharp
+var button = new Button { Text = "Top", HeightRequest = 50 };
+DockLayout.SetDockPosition(button, DockPosition.Top);
+```
+
+Alternatively, `DockLayout` contains a fluent-style `Add(IView, DockPosition)` method that adds a child view to the layout container and sets its docking position.
+
+```csharp
+var dockLayout = new DockLayout();
+var button = dockLayout.Add(new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top);
```

Comment:
```suggestion
dockLayout.Add(new Button { Text = "Top", HeightRequest = 50 }, DockPosition.Top);
```

Reply:
Is the `Add` method no longer fluent?

Reply:
yes, it returns void. We can later change it if necessary, it won't be a breaking change

Reply:
Not sure I get why you made that change?

But at least then you should also change the text accordingly. It says `... contains a fluent-style ...`. Also, the C# code example is no longer useful, it should be changed to

```
var dockLayout = new DockLayout();
var button = new Button { Text = "Top", HeightRequest = 50 };
dockLayout.Add(button, DockPosition.Top);
```

Reply:
I think I may have kickstarted the discussion that resulted in the fluent nature being removed. I got stuck in expecting that the Add would return the DockLayout rather than the newly created child so Adds could be chained together. I suspect we have taken the safest option in not deciding what it should return in the short term as introducing it isn't a breaking change but to remove or change it is.

I will make an effort to remove/update the section about the fluent nature.

Reply:
Thanks for the explanation. Actually my use case was a little different. Usually, after adding a view to the layout, I want to do something with it: e.g. set a binding to another view OR register an event (button pressed) OR the like. But I do see your use case as well and why the "ambiguity" might be an issue ...

Reply:
You might want to checkout our Markup package and it's `Assign` method as it could cover that scenario: https://learn.microsoft.com/en-gb/dotnet/communitytoolkit/maui/markup/extensions/object-extensions#assign